src/monitor.js

```diff
@@ -33,14 +33,23 @@ class Monitor {
       return sentry;
     }).catch(err => {});
 
-    this._sentry.then(sentry => {
+    return this._sentry.then(sentry => {
       sentry.client.captureException(err, {
         tags: _.defaults({
           prefix: this._opts.project + (this._opts.prefix || '.root'),
           process: this._opts.process || 'unknown',
         }, tags),
         level,
       });
+
+      return new Promise((accept, reject) => {
+        sentry.client.once('logged', () => {
+          accept('Succesfully logged error to Sentry.');
+        });
+        sentry.client.once('error', (e) => {
+          reject('Failed to log error to Sentry: ', e.reason);
```

Comment:
Actually, creating a custom error message like this is nice...

But please send error objects back when doing this... Also are you sure there is always an `e.reason` property...
Also `reject` only takes a single parameter afaik...

I would just do: `reject(new Error("Failed sending to sentry: " + err))`